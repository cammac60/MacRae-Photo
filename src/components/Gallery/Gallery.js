import React from 'react';
import { Link } from 'react-router-dom';

import './Gallery.scss';

export const Gallery = props => {
  const { images } = props;
  console.log(images);
  const imagesTiles = images.map((img, i) => {
    const url = `../../images/main/${img.link}.JPG`
    return (
        <img alt={`Photo #${i}`} id={i} className="gallery-tile" src={require(`../../images/main/${img.link}.JPG`)}/>
    )
  });
  return (
    <div className="Gallery">
      <div className="gallery-back">
        <div className="blur-wrapper-2">
          <header className="gallery-header">
            <h2 className="gallery-title">Gallery</h2>
            <div className="gallery-btn-wrapper">
              <Link to="/" className="back-btn" id="gallery-back-btn">Back</Link>
              <a className="ig-link" href="https://www.instagram.com/macrae_5280/?hl=en" target="_blank">
                <img alt="Instagram Logo" src={require('../../images/icons/instagram.svg')} className="ig-logo-2" fill="#ffffff"/>
              </a>
            </div>
          </header>
          <section className="photo-wrapper">
            {imagesTiles}
          </section>
        </div>
      </div>
    </div>
  )
};

export default Gallery;
