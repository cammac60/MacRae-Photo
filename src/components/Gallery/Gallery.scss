@import '../../variables.scss';

.Gallery {
  height: 100vh;
  width: 100vw;
}

.gallery-back {
  background: {
    color: rgba(0, 0, 0, 0.8);
    image: url('../../images/dakota-hogback2.JPG');
    position: center;
    repeat: no-repeat;
    size: cover;
  }
  background-blend-mode: darken;
  height: 100%;
  position: fixed;
  width: 100%;
}

.blur-wrapper-2 {
  backdrop-filter: blur(2px);
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  padding: {
    bottom: 0;
    left: 0;
    right: 0;
    top: 5%;
  }
  height: 100%;
  width: 100%;
}

.gallery-header {
  background: {
    color: rgba(0, 0, 0, 0.6);
    image: url('../../images/dakota-hogback2.JPG');
    position: 50% 12%;
    repeat: no-repeat;
  }
  background-blend-mode: darken;
  border: {
    bottom: 1px solid $main-color;
    top: 1px solid $main-color;
  }
  display: flex;
  justify-content: space-between;
  margin-bottom: auto;
  padding: {
    left: 2%;
    right: 2%;
  }
}

.gallery-title {
  color: $main-accent-color;
  font: {
    family: $large-font;
    size: 6rem;
    weight: normal;
  }
}

.gallery-btn-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 100%;
}

#gallery-back-btn {
  &:hover {
    color: $main-accent-color;
  }
  margin: 0;
}

.ig-logo-2 {
  cursor: pointer;
  height: 50px;
  margin: {
    bottom: 5%;
    left: auto;
  }
  width: 50px;
}

.photo-wrapper {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-column-gap: 20px;
  grid-row-gap: 40px;
  height: 100%;
  margin: {
    left: auto;
    right: auto;
  }
  overflow-x: scroll;
  padding: {
    bottom: 5%;
    top: 5%;
  }
  width: 96%;
}

.gallery-tile {
  border: 1px solid $main-color;
  cursor: pointer;
  height: 400px;
  margin: {
    left: auto;
    right: auto;
  }
  object-fit: cover;
  width: 400px;
  &:active {
    border: 1px solid $main-accent-color;
  }
  &:hover {
    filter: blur(2px);
  }
}
