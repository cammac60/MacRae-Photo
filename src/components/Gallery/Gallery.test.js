import React from 'react';
import { shallow } from 'enzyme';
import Gallery from './Gallery';

describe('Gallery', () => {

  let wrapper, instance;
  const handleImgClick = jest.fn();
  const mockProps = {
    handleImgClick,
    images: [{
      title: 'test',
      link: '25'
    }]
  };

  beforeEach(() => {
    wrapper = shallow(<Gallery {...mockProps}/>);
    instance = wrapper.instance();
  });

  it('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should invoke handleImgClick onClick of the image', () => {
    wrapper.find('.gallery-tile').simulate('click', {});
    expect(handleImgClick).toHaveBeenCalled();
  });

});
