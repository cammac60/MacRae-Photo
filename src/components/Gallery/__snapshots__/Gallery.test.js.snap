// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Should match the snapshot 1`] = `ShallowWrapper {}`;
